wordpress-mobile/WordPress-iOS
# Small Reader Improvements: Present related posts in a new nav stack

Part of #15942 

## Description
- Updated related posts cell to present related posts in a new navigation stack when tapped

## Merge instructions
- Merge after https://github.com/wordpress-mobile/WordPress-iOS/pull/16004

## How to test
1. Go to the reader tab
2. Tap on a reader post and open post details
3. Scroll to the bottom of the post then tap on a related post
5. ✅ The related post should be presented modally
6. Scroll to the bottom of the post then tap on a related post
7. Tap on the back button
8. ✅ The related post from `step 4` should be be displayed
9. Swipe down to dismiss the modal
10. ✅ The original post from `step 2` should be displayed

### Testing analytics
1. Go to the reader tab
2. Tap on a post
3. Scroll to the bottom of the post to see the related posts
4. Tap on a post located in the "More from _BLOG_NAME" where the blog name the the name of the blog for the current post
5. You should see `reader_related_post_from_same_site_clicked`  tracked:
```🔵 Tracked: reader_related_post_from_other_site_clicked <blog_id: BLOG_ID, is_jetpack: IS_JETPACK, offline_view: OFFLINE, post_detail_type: TYPE, post_id: POST_ID>```
5. Go back and tap on a post in the 'More from WordPress.com' section
6. You should see `reader_related_post_from_other_site_clicked`  tracked:
```🔵 Tracked: reader_related_post_from_other_site_clicked <blog_id: BLOG_ID, is_jetpack: IS_JETPACK, offline_view: OFFLINE, post_detail_type: TYPE, post_id: POST_ID>```

## Video


https://user-images.githubusercontent.com/6711616/110054024-b4b48900-7d9d-11eb-89e0-fce2dbea8e70.mov


## PR submission checklist
